packages/account-abstraction/src/SmartAccountAPI.ts

```diff
@@ -84,15 +87,18 @@ export class SmartAccountAPI extends BaseAccountAPI {
     readonly index = 0,
     overheads?: Partial<GasOverheads>
   ) {
+    // Note: todo: clientConfig could itself have PaymasterAPI instead of paymasterUrl or dappAPIKey
     super(provider, entryPoint, clientConfig, accountAddress, overheads)
-    if (!clientConfig.dappAPIKey || clientConfig.dappAPIKey === '') {
+    if (!clientConfig.paymasterUrl || clientConfig.paymasterUrl === '') {
       this.paymasterAPI = undefined
     } else if (clientConfig.customPaymasterAPI) {
       this.paymasterAPI = clientConfig.customPaymasterAPI
     } else {
-      this.paymasterAPI = new BiconomyPaymasterAPI({
-        signingServiceUrl: clientConfig.biconomySigningServiceUrl,
-        dappAPIKey: clientConfig.dappAPIKey,
+      // TODO: notice here it would either directly accept the instance
```

Comment:
@tomarsachin2271 @talhamalik883 we could keep some discussions on this PR itself